### PR TITLE
kconfig: Replace non-defconfig single-symbol 'if's with 'depends on'

### PR DESCRIPTION
**Pull request by ulfalizer**
Same deal as in commit eddd98f811 ("kconfig: Replace some single-symbol
'if's with 'depends on'"), for the remaining cases outside defconfig
files. See that commit for an explanation.

Will do the defconfigs separately in case there are any complaints there.

### CODE REVIEW

**Comment by ulfalizer** [comment]
Will probably always time out.

**Review by henrikbrixandersen — APPROVED** [review]
LGTM.

File: drivers/gpio/Kconfig.stm32 [hunk 1]
```diff
@@ -45,10 +45,9 @@ config GPIO_STM32_PORTJ
 config GPIO_STM32_PORTK
 	bool "Enable GPIO port K support"
 
-if SOC_SERIES_STM32F1X
```
Comment (erwango) [inline drivers/gpio/Kconfig.stm32]: This change is abusive.
There are 5 symbols under the if statement.

**Review by erwango — CHANGES_REQUESTED** [review]
One abusive change.

**Comment by joerchan** [comment]
Looks ok, but are there any advantages to doing this? In case you want to add new kconfigs the ifdef you would have to undo the depends on probably.

File: drivers/gpio/Kconfig.stm32 (hunk 1 above)
Comment (ulfalizer) [reply drivers/gpio/Kconfig.stm32]: The `if` is around a `choice`, so it's fine. These two turn into the same thing after parsing:

```
if FOO

choice
	prompt "choice"

...

endchoice

endif
```

```
choice
	prompt "choice"
	depends on FOO

...

endchoice
```

All `if`s are removed and turned into `depends on` by the code.

**Comment by ulfalizer** [comment]
> Looks ok, but are there any advantages to doing this?

Makes the code (sometimes much) shorter and avoids confusing people re. `depends on` vs. `if` is why I did it.

> In case you want to add new kconfigs the ifdef you would have to undo the depends on probably.

Yup. Don't think it's worthwhile to use the long form everywhere just to give slightly smaller Git diffs though.

**Review by vanti — APPROVED** [review]

File: drivers/gpio/Kconfig.stm32 (hunk 1 above)
Comment (erwango) [reply drivers/gpio/Kconfig.stm32]: My bad I did not see all the symbols were part of the choice.
This is fine then

**Review by erwango — APPROVED** [review]

**Comment by galak** [comment]
Merging this a CI will always timeout due to large number of changes.

**Pull request merged**